class.jetpack-gutenberg.php

```diff
@@ -299,46 +230,49 @@ public static function get_preset( $preset ) {
 	}
 
 	/**
-	 * Filters the results of `apply_filter( 'jetpack_set_available_blocks', array() )`
-	 * using the merged contents of `blocks-manifest.json` ( $preset_blocks )
-	 * and self::$jetpack_blocks ( $internal_blocks )
-	 *
-	 * @param $blocks The default list.
+	 * Returns a whitelist of Jetpack Gutenberg extensions (blocks and plugins), based on index.json
 	 *
 	 * @return array A list of blocks: eg [ 'publicize', 'markdown' ]
 	 */
-	public static function jetpack_set_available_blocks( $blocks ) {
-		$preset_blocks_manifest =  self::preset_exists( 'index' ) ? self::get_preset( 'index' ) : (object) array( 'blocks' => $blocks );
-		// manifest shouldn't remove default blocks...
+	public static function get_jetpack_gutenberg_extensions_whitelist() {
+		$preset_extensions_manifest = self::preset_exists( 'index' ) ? self::get_preset( 'index' ) : (object) array();
 
-		$preset_blocks = isset( $preset_blocks_manifest->production ) ? (array) $preset_blocks_manifest->production : array() ;
-		$preset_blocks = array_unique( array_merge( $preset_blocks, $blocks ) );
+		$preset_extensions = isset( $preset_extensions_manifest->production ) ? (array) $preset_extensions_manifest->production : array() ;
 
 		if ( Jetpack_Constants::is_true( 'JETPACK_BETA_BLOCKS' ) ) {
-			$beta_blocks = isset( $preset_blocks_manifest->beta ) ? (array) $preset_blocks_manifest->beta : array();
-			return array_unique( array_merge( $preset_blocks, $beta_blocks ) );
+			$beta_extensions = isset( $preset_extensions_manifest->beta ) ? (array) $preset_extensions_manifest->beta : array();
+			return array_unique( array_merge( $preset_extensions, $beta_extensions ) );
 		}
 
-		return $preset_blocks;
-	}
-
-	/**
-	 * @deprecated
-	 * @return array A list of block-availability information, eg: [ "publicize" => ["available" => true ], "markdown" => [ "available" => false, "unavailable_reason" => 'missing_module' ] ]
-	 */
-	public static function get_block_availability() {
-		return self::get_availability();
+		return $preset_extensions;
 	}
 
 	/**
 	 * @return array A list of block and plugins and their availablity status
 	 */
 	public static function get_availability() {
-		if ( ! self::should_load() ) {
-			return array();
+		$available_extensions = [];
```

Comment:
This array syntax is failing PHP 5.2 tests

Reply:
0222a32